File: ruby/tests/303-content-userfile.rb
admin = admin_handle
raise "SKIPPED: You have no userfile to handle"             unless @userfile_id_302.present?
raise "SKIPPED: You have no information about file content" unless @one_line_302.present?
raise "SKIPPED: Do not have size of the originale file"     unless @userfile_size_302.present?

##################################
# Stream file (content_userfile) #
##################################

nb_segment = 0
full_size  = 0
admin.content_userfile(@userfile_id_302) do |segment|
  raise "CONTENT USERFILE: segment seems to be inconsistent with the original userfile" if nb_segment == 0 && segment !~ /^#{@one_line_302}/
  full_size  += segment.size
  nb_segment += 1
end

puts  "NUMBER OF SEGMENT=\n#{nb_segment} segment(s)" if @debug
raise "CONTENT USERFILE: less than one segment was streamed" if nb_segment < 2
puts  "SIZE=\n#{full_size}" if @debug
raise "CONTENT USERFILE: size is inconsistent" if full_size != @userfile_size_302

true

File: perl/tests/303-content-userfile.pl
my  $admin = &AdminHandle;
our $userfile_id_302;
our $one_line_302;
our $userfile_size_302;

die "SKIPPED: You have no userfile to handle"             unless defined($userfile_id_302);
die "SKIPPED: You have no information about file content" unless defined($one_line_302);
die "SKIPPED: Do not have size of the original file"      unless defined($userfile_size_302);

##################################
# Stream file (content_userfile) #
##################################

my $nb_segment = 0;
my $full_size  = 0;
my $callback   = sub {
  my ( $chunk, $res, $proto ) = @_;
  die "CONTENT USERFILE: segment seems to be inconsistent with the original userfile\n" if $nb_segment == 0 && $chunk !~  /^${one_line_302}/;
  $full_size  += length($chunk);
  $nb_segment += 1;
};

$admin->content_userfile($userfile_id_302, $callback);

print  "NUMBER OF SEGMENT=\n${nb_segment} segment(s)\n" if $DEBUG;
die "CONTENT USERFILE: less than one segment was streamed\n" if $nb_segment < 2;
print  "SIZE=\n${full_size}\n" if $DEBUG;
die "CONTENT USERFILE: size is inconsistent\n" if $full_size != $userfile_size_302;

1;
